File: backend/api/app/schemas/shop.py
"""
商品相关的 Pydantic 模式
"""
from pydantic import BaseModel, Field
from typing import Optional, List


class WheelSpecResponse(BaseModel):
    """轮毂规格响应"""
    spec_id: int
    size: str
    diameter: Optional[str] = None
    width: Optional[str] = None
    pcd: Optional[str] = None
    offset: Optional[str] = None
    center_bore: Optional[str] = None
    price: Optional[float] = None
    stock: int = 0


class WheelProductResponse(BaseModel):
    """轮毂商品响应"""
    product_id: int
    name: str
    brand_id: Optional[int] = None
    brand_name: Optional[str] = None
    image: Optional[str] = None
    sale_price: Optional[float] = None
    original_price: Optional[float] = None
    price_per: Optional[str] = None
    stock: int = 0
    status: str
    specs: List[WheelSpecResponse] = Field(default_factory=list)


class WheelsListResponse(BaseModel):
    """轮毂商品列表响应"""
    items: List[WheelProductResponse] = Field(default_factory=list)
    total: int = 0
    page: int = 1
    page_size: int = 20

File: frontend/lib/api/shop.ts
/**
 * 商品 API
 * 游客可访问，无需鉴权
 */
import { apiClient } from './client'

export interface WheelSpec {
  spec_id: number
  size: string
  diameter: string | null
  width: string | null
  pcd: string | null
  offset: string | null
  center_bore: string | null
  price: number | null
  stock: number
}

export interface WheelProduct {
  product_id: number
  name: string
  brand_id: number | null
  brand_name: string | null
  image: string | null
  sale_price: number | null
  original_price: number | null
  price_per: string | null
  stock: number
  status: string
  specs: WheelSpec[]
}

export interface WheelsListResponse {
  items: WheelProduct[]
  total: number
  page: number
  page_size: number
}

export interface GetWheelsParams {
  vehicle_id?: string
  pcd?: string
  diameter?: number
  page?: number
  page_size?: number
}

/**
 * 获取轮毂商品列表
 */
export async function getWheels(params: GetWheelsParams = {}): Promise<WheelsListResponse> {
  const searchParams = new URLSearchParams()
  
  if (params.vehicle_id) {
    searchParams.append('vehicle_id', params.vehicle_id)
  }
  if (params.pcd) {
    searchParams.append('pcd', params.pcd)
  }
  if (params.diameter) {
    searchParams.append('diameter', params.diameter.toString())
  }
  if (params.page) {
    searchParams.append('page', params.page.toString())
  }
  if (params.page_size) {
    searchParams.append('page_size', params.page_size.toString())
  }
  
  const queryString = searchParams.toString()
  return apiClient.get<WheelsListResponse>(`/shop/wheels${queryString ? `?${queryString}` : ''}`)
}


